### PR TITLE
Extend type information in tfgen.

### PR DESCRIPTION
**Pull request by pgavlin**
Create rich type information during the gather phase. This type
information can then be used to drive code generation with fewer special
cases.

### CODE REVIEW

File: pkg/tfgen/generate.go [hunk 1]
```diff
@@ -201,6 +201,125 @@ type moduleMember interface {
...
+	element    *propertyType
+	properties []*variable
+
+	typ        tokens.Type
+	nestedType tokens.Type
+	altTypes   []tokens.Type
+	asset      *tfbridge.AssetTranslation
+}
+
+func makePropertyType(sch *schema.Schema, info *tfbridge.SchemaInfo, out bool, parsedDocs parsedDoc) *propertyType {
```
Comment (pgavlin) [inline pkg/tfgen/generate.go]: This centralizes the handling of `MaxItemsOne`, `TypeMap` with a `*schema.Resource` element, and doc attachment. These changes provide the most important simplifications in the downstream code.

**Comment by jen20** [comment]
I've rebased this on top of #50 now that has been merged - this should run the downstream tests and give us confidence in the generation changes.

**Comment by pgavlin** [comment]
@jen20 how do I see the generated source code and check for diffs?

**Comment by jen20** [comment]
@pgavlin Right now you look at each job in the actions tab, and the diff is presented at the end of the "Test Downstream" step. In the fairly short term it will post a link to the GitHub diff back to the PR.

**Comment by pgavlin** [comment]
Thanks! Looks like this is clean, then. 👍

**Pull request merged**